test/eth/next_interval_calculator.spec.js

```diff
@@ -1,189 +1,177 @@
+  it('interval is correct if the blockchain node is ahead', async () => {
+    const worker = new eth_worker.worker(constants);
 
-        worker.lastExportedBlock = 1;
-        worker.lastConfirmedBlock = constants.BLOCK_INTERVAL + constants.CONFIRMATIONS + 10;
+    worker.lastExportedBlock = 1;
+    worker.lastConfirmedBlock = 2;
 
-        const result = await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(result.success, true);
-        assert.deepStrictEqual(result.fromBlock, 2);
-        assert.deepStrictEqual(result.toBlock, 1 + constants.BLOCK_INTERVAL);
-    });
+    const context = await analyzeWorkerContext(worker);
+    setWorkerSleepTime(worker, context);
+    const result = nextIntervalCalculator(worker);
 
+    assert.deepStrictEqual(context, WORK_NO_SLEEP);
+    assert.deepStrictEqual(worker.sleepTimeMsec, 0);
+    assert.deepStrictEqual(result.fromBlock, 2);
+    assert.deepStrictEqual(result.toBlock, 2);
 });
 
-describe('Check logic when Node is not ahead', function () {
-    it('Check interval is correct if Node is behind', async function () {
-        const worker = new eth_worker.worker(constants);
-        worker.web3Wrapper = new MockWeb3Wrapper(constants.CONFIRMATIONS + 10);
-
-        worker.lastExportedBlock = 2;
-        worker.lastConfirmedBlock = 1;
+  it('interval is correct if the blockchain node is behind', async () => {
+    const worker = new eth_worker.worker(constants);
+    worker.web3Wrapper = new MockWeb3Wrapper(constants.CONFIRMATIONS + 10);
 
-        const resultSecond = await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(resultSecond.success, true);
-        assert.deepStrictEqual(resultSecond.fromBlock, 3);
-        assert.deepStrictEqual(resultSecond.toBlock, 10);
-    });
+    worker.lastExportedBlock = 2;
+    worker.lastConfirmedBlock = 1;
 
-    it('Sleep time set if Node is not ahead', async function () {
-        const worker = new eth_worker.worker(constants);
-        worker.web3Wrapper = new MockWeb3Wrapper(constants.CONFIRMATIONS + 10);
+    const context = await analyzeWorkerContext(worker);
+    setWorkerSleepTime(worker, context);
+    const result = nextIntervalCalculator(worker);
 
-        worker.lastExportedBlock = 1;
-        worker.lastConfirmedBlock = 1;
+    assert.deepStrictEqual(context, WORK_SLEEP);
+    assert.deepStrictEqual(worker.sleepTimeMsec, constants.LOOP_INTERVAL_CURRENT_MODE_SEC * 1000);
+    assert.deepStrictEqual(result.fromBlock, 3);
+    assert.deepStrictEqual(result.toBlock, 10);
+  });
 
-        await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(worker.sleepTimeMsec, constants.LOOP_INTERVAL_CURRENT_MODE_SEC * 1000);
-    });
+  it('node is not called', async () => {
+    const worker = new eth_worker.worker(constants);
+    worker.web3Wrapper = new MockWeb3Wrapper();
+    worker.lastExportedBlock = 1;
+    worker.lastConfirmedBlock = 2;
 
-    it('Node gets called if Node is not ahead', async function () {
```

Comment:
This test is similar to the one below and tests implementation. Maybe it makes sense to restore it.